### PR TITLE
update `BarcodeDetector` exceptions and parameters

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

the `BarcodeDetector` constructor may throw `TypeError` in some cases, see https://wicg.github.io/shape-detection-api/#barcode-detection-api, and its parameter `formats` is also optional

the `BarcodeDetector.detect()` method may throw `SecurityError` and `InvalidStateError` when check image source, see https://wicg.github.io/shape-detection-api/#image-sources-for-detection

### CODE REVIEW

File: files/en-us/web/api/barcodedetector/detect/index.md [hunk 1]
```diff
@@ -51,6 +49,10 @@ Returns a {{jsxref('Promise')}} which fulfills with an array of
 - {{jsxref("TypeError")}}
   - : No parameter is specified or the `type` is not that of an
     `ImageBitmapSource`.
+- `SecurityError` {{domxref("DOMException")}}
+  - : Thrown if the `imageBitmapSource` has an origin and is not the same as the document's origin, or if the `imageBitmapSource` is a {{domxref('HTMLCanvasElement')}} and its [origin-clean](https://html.spec.whatwg.org/multipage/canvas.html#concept-canvas-origin-clean) flag it set to `false`.
```
Comment (bsmth) [inline files/en-us/web/api/barcodedetector/detect/index.md:53]:
```suggestion
  - : Thrown if the `imageBitmapSource` has an origin and is not the same as the document's origin, or if the `imageBitmapSource` is a {{domxref('HTMLCanvasElement')}} and its [origin-clean](https://html.spec.whatwg.org/multipage/canvas.html#concept-canvas-origin-clean) flag is set to `false`.
```

File: files/en-us/web/api/barcodedetector/barcodedetector/index.md [hunk 2]
```diff
@@ -29,10 +29,15 @@ new BarcodeDetector(options)
     in the subsequent {{domxref('BarcodeDetector.detect()','detect()')}} calls. The
     options are:
 
-    - `formats`
+    - `formats` {{optional_inline}}
       - : An {{jsxref('Array')}} of barcode formats as strings.
         To see a full list of supported formats see the [Barcode Detection API](/en-US/docs/Web/API/Barcode_Detection_API).
```
Comment (bsmth) [inline files/en-us/web/api/barcodedetector/barcodedetector/index.md:34]: I feel like this is useful to know:

```suggestion
        If not provided, `detect()` calls search for all supported formats.
        Limiting to specific formats is therefore recommended for performance reasons.
        To see a full list of supported formats see the [Barcode Detection API](/en-US/docs/Web/API/Barcode_Detection_API).
```

File: files/en-us/web/api/barcodedetector/detect/index.md [hunk 3]
```diff
@@ -51,6 +49,10 @@ Returns a {{jsxref('Promise')}} which fulfills with an array of
 - {{jsxref("TypeError")}}
   - : No parameter is specified or the `type` is not that of an
     `ImageBitmapSource`.
+- `SecurityError` {{domxref("DOMException")}}
+  - : Thrown if the `imageBitmapSource` has an origin and is not the same as the document's origin, or if the `imageBitmapSource` is a {{domxref('HTMLCanvasElement')}} and its [origin-clean](https://html.spec.whatwg.org/multipage/canvas.html#concept-canvas-origin-clean) flag is set to `false`.
+- `InvalidStateError` {{domxref("DOMException")}}
+  - : Thrown if the `imageBitmapSource` is an {{domxref('HTMLImageElement')}} and it is not fully decoded or decodes failed, or if the `imageBitmapSource` is an {{domxref('HTMLVideoElement')}} and its {{domxref('HTMLMediaElement.readyState', 'readyState')}} is `HAVE_NOTHING` or `HAVE_METADATA`.
```
Comment (bsmth) [inline files/en-us/web/api/barcodedetector/detect/index.md:55]: There's a corrupted (https://html.spec.whatwg.org/multipage/images.html#img-error) state, but this is implied by decoding failed, so I think this covers it:

```suggestion
  - : Thrown if the `imageBitmapSource` is an {{domxref('HTMLImageElement')}} and is not fully decoded or decoding failed, or is an {{domxref('HTMLVideoElement')}} and its {{domxref('HTMLMediaElement.readyState', 'readyState')}} is `HAVE_NOTHING` or `HAVE_METADATA`.
```

File: files/en-us/web/api/barcodedetector/barcodedetector/index.md [hunk 4]
```diff
@@ -29,10 +29,17 @@ new BarcodeDetector(options)
     in the subsequent {{domxref('BarcodeDetector.detect()','detect()')}} calls. The
     options are:
 
-    - `formats`
+    - `formats` {{optional_inline}}
       - : An {{jsxref('Array')}} of barcode formats as strings.
+        If not provided, `detect()` calls search for all supported formats.
+        Limiting to specific formats is therefore recommended for performance reasons.
         To see a full list of supported formats see the [Barcode Detection API](/en-US/docs/Web/API/Barcode_Detection_API).
```
Comment (skyclouds2001) [inline files/en-us/web/api/barcodedetector/barcodedetector/index.md:36]:
```suggestion
        To see a full list of supported formats see the {{domxref("Barcode Detection API")}}.
```

**Review by bsmth — APPROVED** [review]
Thanks a lot! 🙌🏻

**Pull request merged**